mm2src/lp_ordermatch.rs

```diff
@@ -1547,54 +1656,281 @@ enum OrderbookRequestingState {
     NotRequested { subscribed_at: u64 },
 }
 
+type H64 = [u8; 8];
+
+struct TrieDiff<Key, Value> {
+    delta: Vec<(Key, Option<Value>)>,
+    next_root: H64,
+}
+
+impl<Key: std::fmt::Debug, Value: std::fmt::Debug> std::fmt::Debug for TrieDiff<Key, Value> {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.debug_struct("TrieDiff")
+            .field("delta", &self.delta)
+            .field("next_root", &self.next_root)
+            .finish()
+    }
+}
+
+impl<Key: PartialEq, Value: PartialEq> PartialEq for TrieDiff<Key, Value> {
+    fn eq(&self, other: &TrieDiff<Key, Value>) -> bool {
+        self.delta == other.delta && self.next_root == other.next_root
+    }
+}
+
+impl<Key: Eq, Value: Eq> Eq for TrieDiff<Key, Value> {}
+
+#[derive(Eq, PartialEq)]
+struct TrieDiffHistory<Key, Value> {
+    inner: HashMap<H64, TrieDiff<Key, Value>>,
+}
+
+impl<Key, Value> Default for TrieDiffHistory<Key, Value> {
```

Comment:
The same comment

Reply:
The same response :slightly_smiling_face:

Reply:
The Default case is a bit different btw, following happens if I derive Default for TrieDiffHistory:
```
error[E0277]: the trait bound `lp_ordermatch::OrderbookItem: Default` is not satisfied
    --> mm2src/lp_ordermatch.rs:1718:5
     |
1718 |     order_pairs_trie_state_history: TrieDiffHistory<Uuid, OrderbookItem>,
     |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `Default` is not implemented for `lp_ordermatch::OrderbookItem`
     |
     = note: required because of the requirements on the impl of `Default` for `lp_ordermatch::TrieDiffHistory<uuid::Uuid, lp_ordermatch::OrderbookItem>`
     = note: required by `std::default::Default::default`
     = note: this error originates in a derive macro (in Nightly builds, run with -Z macro-backtrace for more info)
```

Reply:
Ok, I got :)

Reply:
It is a known issue for long time: https://stackoverflow.com/a/46080328, the `Default` bound is not really required for `OrderbookItem` here so manual implementation of `Default` for `TrieDiffHistory` seems the only workaround.